### PR TITLE
ZM clean up round 2

### LINKED ISSUE #83: Change directory name from zm to zhang_mcfarlane
### What is the feature/what would you like to discuss?

@swrneale requested that the name of the directory should be zhang_mcfarlane instead of zm.  This name will be more descriptive and allow folks to understand what is in this directory.

### Is there anyone in particular you want to be part of this conversation?

_No response_

### Will this change (regression test) answers?

No

### Will you be implementing this enhancement yourself?

Yes

### PR DESCRIPTION
**Pull request by cacraigucar**
More cleanup identified during CCPP standard name meetings

Includes preliminary metadata for ZM (ignore for now as I know it is not complete - next review should include the entire files)

Closes #83

### CODE REVIEW

File: zhang_mcfarlane/zm_conv_momtran.F90 [hunk 1]
```diff
@@ -111,31 +116,34 @@ subroutine zm_conv_momtran_run(ncol, pver, pverp, &
...
 
-   real(kind_phys),intent(out) ::  pguall(:,:,:)      ! Apparent force from  updraft PG   ! (ncol,pver,ncnst)
-   real(kind_phys),intent(out) ::  pgdall(:,:,:)      ! Apparent force from  downdraft PG ! (ncol,pver,ncnst)
+   real(kind_phys),intent(out) ::  pguall(:,:,:)      ! Apparent force from  updraft PG   ! (ncol,pver,2)
+   real(kind_phys),intent(out) ::  pgdall(:,:,:)      ! Apparent force from  downdraft PG ! (ncol,pver,2)
 
-   real(kind_phys),intent(out) ::  icwu(:,:,:)      ! In-cloud winds in updraft           ! (ncol,pver,ncnst)
-   real(kind_phys),intent(out) ::  icwd(:,:,:)      ! In-cloud winds in downdraft         ! (ncol,pver,ncnst)
+   real(kind_phys),intent(out) ::  icwu(:,:,:)      ! In-cloud winds in updraft           ! (ncol,pver,2)
+   real(kind_phys),intent(out) ::  icwd(:,:,:)      ! In-cloud winds in downdraft         ! (ncol,pver,2)
```
Comment (cacraigucar) [inline zhang_mcfarlane/zm_conv_momtran.F90:123]: Neglected to separate these into two - need to do this

Comment (cacraigucar) [reply zhang_mcfarlane/zm_conv_momtran.F90]: This has now been done

File: zhang_mcfarlane/zm_conv_momtran.F90 [hunk 2]
```diff
@@ -111,31 +115,45 @@ subroutine zm_conv_momtran_run(ncol, pver, pverp, &
    real(kind_phys) pgu(ncol,pver)      ! Pressure gradient term for updraft
    real(kind_phys) pgd(ncol,pver)      ! Pressure gradient term for downdraft
 
-   real(kind_phys),intent(out) ::  pguall(:,:,:)      ! Apparent force from  updraft PG   ! (ncol,pver,ncnst)
-   real(kind_phys),intent(out) ::  pgdall(:,:,:)      ! Apparent force from  downdraft PG ! (ncol,pver,ncnst)
+   real(kind_phys),intent(out) ::  pguallu(:,:)      ! Apparent force from  updraft PG U winds  ! (ncol,pver)
+   real(kind_phys),intent(out) ::  pguallv(:,:)      ! Apparent force from  updraft PG V winds  ! (ncol,pver)
+   real(kind_phys),intent(out) ::  pgdallu(:,:)      ! Apparent force from  downdraft PG U winds! (ncol,pver)
+   real(kind_phys),intent(out) ::  pgdallv(:,:)      ! Apparent force from  downdraft PG V winds! (ncol,pver)
```
Comment (nusbaume) [inline zhang_mcfarlane/zm_conv_momtran.F90:121]: Not to be grammar-obsessed, but:

```suggestion
   real(kind_phys),intent(out) ::  pguallu(:,:)      ! Apparent force from  updraft PG on U winds  ! (ncol,pver)
   real(kind_phys),intent(out) ::  pguallv(:,:)      ! Apparent force from  updraft PG on V winds  ! (ncol,pver)
   real(kind_phys),intent(out) ::  pgdallu(:,:)      ! Apparent force from  downdraft PG on U winds! (ncol,pver)
   real(kind_phys),intent(out) ::  pgdallv(:,:)      ! Apparent force from  downdraft PG  on V winds! (ncol,pver)
```

File: zhang_mcfarlane/zm_conv_momtran.F90 [hunk 3]
```diff
@@ -111,31 +115,45 @@ subroutine zm_conv_momtran_run(ncol, pver, pverp, &
...
+   real(kind_phys),intent(out) ::  icwdv(:,:)      ! In-cloud V winds in downdraft         ! (ncol,pver)
 
    real(kind_phys),intent(out) ::  seten(:,:) ! Dry static energy tendency                ! (ncol,pver)
    real(kind_phys)                 gseten(ncol,pver) ! Gathered dry static energy tendency
 
-   real(kind_phys)  mflux(ncol,pverp,ncnst)   ! Gathered momentum flux
+   real(kind_phys) :: winds(ncol,pver,num_winds)       ! combined winds array
+   real(kind_phys) :: wind_tends(ncol,pver,num_winds)  ! combined tendency array
+   real(kind_phys) :: pguall(ncol,pver,num_winds)      ! Combined apparent force from  updraft PG U winds
+   real(kind_phys) :: pgdall(ncol,pver,num_winds)      ! Combined apparent force from  downdraft PG U winds
```
Comment (nusbaume) [inline zhang_mcfarlane/zm_conv_momtran.F90:134]: Comment grammar:

```suggestion
   real(kind_phys) :: pguall(ncol,pver,num_winds)      ! Combined apparent force from  updraft PG  on winds
   real(kind_phys) :: pgdall(ncol,pver,num_winds)      ! Combined apparent force from  downdraft PG on winds
```

File: zhang_mcfarlane/zm_conv_momtran.F90 [hunk 4]
```diff
@@ -416,6 +434,19 @@ subroutine zm_conv_momtran_run(ncol, pver, pverp, &
...
+! Split out the wind tendencies
+   windu_tend(:,:) = wind_tends(:,:,1)
+   windv_tend(:,:) = wind_tends(:,:,2)
+
+   pguallu(:,:)     = pguall(:,:,1)
+   pguallv(:,:)     = pguall(:,:,2)
+   pgdallu(:,:)     = pgdall(:,:,1)
+   pgdallv(:,:)     = pgdall(:,:,2)
+   icwuu(:ncol,:)       = icwu(:,:,1)
+   icwuu(:ncol,:)       = icwu(:,:,1)
```
Comment (nusbaume) [inline zhang_mcfarlane/zm_conv_momtran.F90:446]: I think this needs to be:

```suggestion
   icwuv(:ncol,:)       = icwu(:,:,2)
```

**Review by nusbaume — CHANGES_REQUESTED** [review]
Looks good!  Just a few changes requested in the momtran code.

Also I didn't review the metadata files, but if you would like me to at this stage please let me know.  Thanks!

File: zhang_mcfarlane/zm_conv_momtran.F90 (hunk 2 above)
Comment (cacraigucar) [reply zhang_mcfarlane/zm_conv_momtran.F90:121]: fixed

File: zhang_mcfarlane/zm_conv_momtran.F90 (hunk 3 above)
Comment (cacraigucar) [reply zhang_mcfarlane/zm_conv_momtran.F90:134]: fixed - bonus fix - corrected "comibined" a right below

File: zhang_mcfarlane/zm_conv_momtran.F90 (hunk 4 above)
Comment (cacraigucar) [reply zhang_mcfarlane/zm_conv_momtran.F90:446]: good catch!

File: zhang_mcfarlane/zm_conv_momtran.F90 [hunk 5]
```diff
@@ -130,10 +130,10 @@ subroutine zm_conv_momtran_run(ncol, pver, pverp, &
 
    real(kind_phys) :: winds(ncol,pver,num_winds)       ! combined winds array
    real(kind_phys) :: wind_tends(ncol,pver,num_winds)  ! combined tendency array
-   real(kind_phys) :: pguall(ncol,pver,num_winds)      ! Combined apparent force from  updraft PG U winds
-   real(kind_phys) :: pgdall(ncol,pver,num_winds)      ! Combined apparent force from  downdraft PG U winds
-   real(kind_phys) :: icwu(ncol,pver,num_winds)        ! Comibined In-cloud winds in updraft
-   real(kind_phys) :: icwd(ncol,pver,num_winds)        ! Comibined In-cloud winds in downdraft
+   real(kind_phys) :: pguall(ncol,pver,num_winds)      ! Combined apparent force from  updraft on PG U winds
+   real(kind_phys) :: pgdall(ncol,pver,num_winds)      ! Combined apparent force from  downdraft on PG U winds
```
Comment (nusbaume) [inline zhang_mcfarlane/zm_conv_momtran.F90:134]: I think `on` should be after `PG` here:

```suggestion
   real(kind_phys) :: pguall(ncol,pver,num_winds)      ! Combined apparent force from  updraft PG on U winds
   real(kind_phys) :: pgdall(ncol,pver,num_winds)      ! Combined apparent force from  downdraft PG on U winds
```

**Review by nusbaume — APPROVED** [review]
I have one last comment change request, but it doesn't require a re-review.  Otherwise everything looks great now, thanks!

**Pull request merged**